packages/vm/src/testing/storage.rs

```diff
@@ -33,22 +36,43 @@ impl ReadonlyStorage for MockStorage {
         start: Option<&[u8]>,
         end: Option<&[u8]>,
         order: Order,
-    ) -> FfiResult<Box<dyn Iterator<Item = FfiResult<KV>> + 'a>> {
+    ) -> FfiResult<(Box<dyn Iterator<Item = StorageIteratorItem> + 'a>, u64)> {
         let bounds = range_bounds(start, end);
 
         // BTreeMap.range panics if range is start > end.
         // However, this cases represent just empty range and we treat it as such.
         match (bounds.start_bound(), bounds.end_bound()) {
             (Bound::Included(start), Bound::Excluded(end)) if start > end => {
-                return Ok(Box::new(iter::empty()));
+                return Ok((Box::new(iter::empty()), 0));
             }
             _ => {}
         }
 
         let iter = self.data.range(bounds);
         Ok(match order {
-            Order::Ascending => Box::new(iter.map(clone_item).map(FfiResult::Ok)),
-            Order::Descending => Box::new(iter.rev().map(clone_item).map(FfiResult::Ok)),
+            Order::Ascending => (
+                Box::new(
+                    iter.map(clone_item)
+                        .map(|item| {
+                            let gas_cost = (item.0.len() + item.1.len()) as u64;
```

Comment:
Ah, nice, non-zero gas used for tests. I like this :smile: 

You could add a large constant if you want to be like the sdk, but this is fine as well. Just that this is non-zero for reads to properly test